File: test-suite/src/test/java/helloworld/DnsClientsTest.java
package helloworld;

import io.micronaut.context.annotation.Property;
import io.micronaut.test.extensions.junit5.annotation.MicronautTest;
import jakarta.inject.Inject;
import org.junit.jupiter.api.Test;

import static org.junit.jupiter.api.Assertions.assertEquals;

@Property(name = "spec.name", value = "DnsClientsTest")
@MicronautTest
class DnsClientsTest {

    @Inject
    GreeterGrpc.GreeterStub greeterStub;

    @Test
    void testDnsTarget() {
        assertEquals("greeter", greeterStub.getChannel().authority());
    }
}

File: test-suite-kotlin/src/test/kotlin/helloworld/DnsClientsTest.kt
package helloworld

import io.micronaut.context.annotation.Property
import io.micronaut.test.extensions.junit5.annotation.MicronautTest
import jakarta.inject.Inject
import org.junit.jupiter.api.Assertions.assertEquals
import org.junit.jupiter.api.Test

@Property(name = "spec.name", value = "DnsClientsTest")
@MicronautTest
class DnsClientsTest {

    @Inject
    lateinit var greeterStub: GreeterGrpc.GreeterStub

    @Test
    fun testDnsTarget() {
        assertEquals("greeter", greeterStub.channel.authority())
    }
}
